packages/remirror__extension-link/src/link-extension.ts

```diff
@@ -205,6 +210,15 @@ export class LinkExtension extends MarkExtension<LinkOptions> {
     return 'link' as const;
   }
 
+  /**
+   * The autoLinkRegex option with the global flag removed, ensure no "lastIndex" state is maintained over multiple matches
+   * @private
+   */
+  private _autoLinkRegexNonGlobal = new RegExp(
+    `^${DEFAULT_AUTO_LINK_REGEX.source}$`,
+    DEFAULT_AUTO_LINK_REGEX.flags.replace('g', ''),
+  );
```

Comment:
Is this needed here, if `init` will anyways update it? The replace might be "icky", but if it is in just one place it could be documented there to stand a chance of survival :)

Reply:
Yep fair point, just added it to avoid a `this. _autoLinkRegexNonGlobal` might be undefined type warning.